x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/SecuritySystemIndices.java

```diff
@@ -683,4 +696,139 @@ private XContentBuilder getTokenIndexMappings() {
+                builder.endObject();
+
+                builder.field("dynamic", "strict");
+                builder.startObject("properties");
+                {
+                    builder.startObject("uid");
+                    builder.field("type", "keyword");
+                    builder.endObject();
+
+                    builder.startObject("redirect_uid");
+                    builder.field("type", "keyword");
+                    builder.endObject();
+
+                    builder.startObject("enabled");
+                    builder.field("type", "boolean");
+                    builder.endObject();
+
+                    builder.startObject("user");
+                    {
+                        builder.field("type", "object");
+                        builder.startObject("properties");
+                        {
+                            builder.startObject("username");
+                            builder.field("type", "keyword");
+                            builder.endObject();
+
+                            builder.startObject("realm");
+                            {
+                                builder.field("type", "object");
+                                builder.startObject("properties");
+                                {
+                                    builder.startObject("name");
+                                    builder.field("type", "keyword");
+                                    builder.endObject();
+
+                                    builder.startObject("type");
+                                    builder.field("type", "keyword");
+                                    builder.endObject();
+
+                                    builder.startObject("domain");
+                                    builder.field("type", "keyword");
+                                    builder.endObject();
+                                }
+                                builder.endObject();
+                            }
+                            builder.endObject();
+
+                            builder.startObject("email");
+                            builder.field("type", "text");
+                            builder.endObject();
+
+                            builder.startObject("full_name");
+                            builder.field("type", "text");
+                            builder.endObject();
+
+                            builder.startObject("display_name");
+                            builder.field("type", "text");
+                            builder.endObject();
+                        }
+                        builder.endObject();
+                    }
+                    builder.endObject();
+
+                    builder.startObject("last_synchronized");
+                    builder.field("type", "date");
+                    builder.field("format", "epoch_millis");
+                    builder.endObject();
+
+                    builder.startObject("access");
```

Comment:
What do you think about `last_login` instead of `access`? Just a suggestion.

Reply:
I pondered on this and my conclusion is to stick with `access`. A few reasons:
* The whole concept of profile in ES has the assumption of last login because currently it is not possible to know anything about users from some external realms. So this assumption is implicit but known.
* Strictly speaking, the fields `email`, `full_name` are also from last login. But I feel it is too verbose to qualify all of them. We may also wanna make special case for `full_name` to not take value from the last login if it is `null`, i.e. we don't overwrite an existing value with `null`. In that case, it's hard to give this field a proper qualifier. 
* We talked about the possibilities to update these fields without the need of login. This is already technically possible today for native realm users. It might be possible in future to do it for more realm types. In these situations, the name `last_login` would become obsolete and inaccurate.

I do agree it is important to call out that these fields mostly reflect the values from last login. But I think we can solve it with documentation and other communications, especially given the consumers of this document are mostly internal.